support/k8s/k8s-workload-registrar/mode-crd/controllers/identity_schema.go

```diff
@@ -0,0 +1,313 @@
+//package main
+package controllers
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+
+	spiffeidv1beta1 "github.com/spiffe/spire/support/k8s/k8s-workload-registrar/mode-crd/api/spiffeid/v1beta1"
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	// attestor variants:
+	nodeAttestor     = "nodeAttestor"
+	workloadAttestor = "workloadAttestor"
+
+	// selector labels
+	namespaceLabel      = "Namespace"
+	podUIDLabel         = "PodUID"
+	podNameLabel        = "PodName"
+	serviceAccountLabel = "ServiceAccount"
+)
+
+type IdentitySchema struct {
+	Version string  `yaml:"version"`
+	Fields  []Field `yaml:"fields"`
+}
+
+type Field struct {
+	Name            string           `yaml:"name"`
+	AttestorSource  *AttestorSource  `yaml:"attestorSource,omitempty"`
+	ConfigMapSource *ConfigMapSource `yaml:"configMapSource,omitempty"`
+}
+
+type Source interface {
+	GetValue(pod *corev1.Pod) (string, error)
+}
+
+type ConfigMapSource struct {
+	Name      string `yaml:"name"`
+	Namespace string `yaml:"ns"`
+	Field     string `yaml:"field"`
+}
+
+type AttestorSource struct {
+	Name    string    `yaml:"name"`
+	Group   string    `yaml:"group"`
```

Comment:
in my view this is either one of `WorkloadAttestor struct` or `NodeAttestor struct`, the reason i see this as 2 distinct objects is that both probably will require separate specific information to be used -  node attestors need SPIRE server access, but not pod access and otherwise for workload attestors.